Avoid starting DNSMASQ when in repeater mode

ratcashdev opened the pull request:
Logs filled with 'watchdog 250:notify_rc start_dnsmasq' every 30 seconds if the device (AC68U) is in repeater mode.
The condition seems to be wrong, fixed.

paldier commented:
in repeater mode, dnsmasq is required when not connected, this is not a bug, please check your /jffs/configs and /jffs/scripts directories, delete the broken soft links or files, or delete all.

matyasberry commented:
Both directories are empty. Maybe the logs are harmless, just found it strange, that I see those lines every 30 secs.
Moreover, connections to this repeater are unstable. After-50-10 mins the connection loses INET and is unable to obtain an IP.
Thought that's because logs fill up space and cause trouble.

ratcashdev commented:
Both directories are empty. Maybe the logs are harmless, just found it strange, that I see those lines every 30 secs.
Moreover, connections to this repeater are unstable. After-50-10 mins the connection loses INET and is unable to obtain an IP.
Thought that's because logs fill up space and cause trouble.

paldier commented:
5 mins? `restart watchdog for no heartbeat`, did you find this message in the log.

matyasberry commented:
The following is a snippet from the log. This snippet repeats exactly every 15 mins.
<details><summary>snippet</summary>

```
Sep 16 13:00:55 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: Adjusted channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: acs_set_chspec: 0x180b (9l) for reason APCS_CSTIMER
Sep 16 13:01:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:01:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:02:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:02:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:03:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:03:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:04:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:04:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:05:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:05:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:06:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:06:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:07:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:07:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:08:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:08:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:09:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:09:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:10:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:10:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:11:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:11:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:12:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:12:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:13:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:13:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:14:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:14:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:59 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: Adjusted channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: acs_set_chspec: 0x180b (9l) for reason APCS_CSTIMER
Sep 16 13:16:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:16:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:17:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:17:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:18:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:18:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:19:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:19:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:20:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:20:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:21:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:21:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:22:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:22:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:23:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:23:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:24:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:24:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:25:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:25:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:26:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:26:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:27:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:27:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:28:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:28:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:29:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:29:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:30:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:30:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:31:03 acsd: selected channel spec: 0x180b (9l)
```
</details>

Here a short snippet with a station leaving (my view is that it was unvoluntary, since that stations is always on and non-moveable):
```
Sep 16 17:08:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 17:08:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 17:09:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 17:09:33 syslog: wlceventd_proc_event(527): eth1: Auth 9C:5A:81:BC:8A:AC, status: Successful (0), rssi:0
Sep 16 17:09:33 syslog: wlceventd_proc_event(556): eth1: Assoc 9C:5A:81:BC:8A:AC, status: Successful (0), rssi:0
Sep 16 17:09:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 17:09:56 syslog: wlceventd_proc_event(491): eth1: Deauth_ind 9C:5A:81:BC:8A:AC, status: 0, reason: Deauthenticated because sending station is leaving (or has left) IBSS or ESS (3), rssi:0
Sep 16 17:10:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 17:10:46 rc_service: watchdog 250:notify_rc start_dnsmasq
```

ratcashdev commented:
The following is a snippet from the log. This snippet repeats exactly every 15 mins.
<details><summary>snippet</summary>

```
Sep 16 13:00:55 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: Adjusted channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:00:55 acsd: acs_set_chspec: 0x180b (9l) for reason APCS_CSTIMER
Sep 16 13:01:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:01:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:02:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:02:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:03:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:03:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:04:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:04:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:05:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:05:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:06:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:06:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:07:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:07:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:08:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:08:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:09:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:09:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:10:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:10:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:11:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:11:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:12:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:12:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:13:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:13:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:14:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:14:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:15:59 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: Adjusted channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: selected channel spec: 0x180b (9l)
Sep 16 13:15:59 acsd: acs_set_chspec: 0x180b (9l) for reason APCS_CSTIMER
Sep 16 13:16:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:16:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:17:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:17:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:18:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:18:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:19:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:19:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:20:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:20:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:21:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:21:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:22:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:22:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:23:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:23:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:24:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:24:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:25:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:25:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:26:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:26:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:27:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:27:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:28:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:28:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:29:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:29:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:30:16 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:30:46 rc_service: watchdog 250:notify_rc start_dnsmasq
Sep 16 13:31:03 acsd: selected channel spec: 0x180b (9l)
```
</details>

I did not find the "restart watchdog" message in the logs.
```
less syslog.log | grep "restart" |wc -l
0
```

ratcashdev commented:
I have realized that start_dnsmasq is not working:
```
admin@RT-AC68U-B6E8:/tmp/home/root# service start_dnsmasq

Done.
admin@RT-AC68U-B6E8:/tmp/home/root# ps |grep dns
 4431 admin     1440 S    grep dns
```

ratcashdev commented:
I just realized that the specific box is RT-AC68U, HW Revision: E1
```
# nvram get HW_ver
1.95
# nvram get boardrev
0x1103
# nvram get bl_version
1.3.0.7
```
